pkgs/applications/misc/paperjam/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib , stdenv, fetchgit, qpdf, libpaper, asciidoc, }:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "paperjam";
+  version = "1.2";
+
+  src = fetchgit {
+    url = "git://git.ucw.cz/paperjam.git";
+    # by recording the exact commit, we can embed it into the binary
+    # this can't be done automatically as fetchgit removes .git directory
+    rev = "90173f436b39bfe39491aa356a218ac185618d1c";
+    hash = "sha256-F0bcYXQhtaI48ULhi7o8gLlC0jIuxPIKTq1cABrP2Iw=";
+  };
+
+  buildInputs = [
+    qpdf
+    libpaper
+    asciidoc
+  ];
+
+  makeFlags = [
+    "DESTDIR=$(out)"
+    "PREFIX=/"
+    # prevent real build date which is impure
+    "BUILD_DATE=\\<unknown\\>"
+    "BUILD_COMMIT=${finalAttrs.src.rev}"
```

Comment:
Is this truly needed?

Reply:
As the comment states, `BUILD_DATE` is. `BUILD_COMMIT` is just a convenience thing for the output of `paperjam --version`.